pulp_rpm/app/downloaders.py

```diff
@@ -12,18 +10,36 @@
 class RpmDownloader(HttpDownloader):
     """
     Custom Downloader that automatically handles authentication token for SLES repositories.
+
+    Args:
+        silence_errors_for_response_status_codes (iterable): An iterable of response exception
+            codes to be ignored when raising exception. e.g. `{404}`
+        sles_auth_token (str): SLES authentication token.
+    Raises:
+        FileNotFoundError: If aiohttp response status is >= 400 and not silenced.
+
     """
 
-    def __init__(self, *args, **kwargs):
+    def __init__(self, *args, silence_errors_for_response_status_codes=set(), sles_auth_token=None,
+                 **kwargs):
         """
         Initialize the downloader.
         """
-        if 'sles_auth_token' in kwargs:
-            self.sles_auth_token = kwargs.pop('sles_auth_token')
-        else:
-            self.sles_auth_token = None
+        self.sles_auth_token = sles_auth_token
+        self.silence_errors_for_response_status_codes = silence_errors_for_response_status_codes
+
         super().__init__(*args, **kwargs)
 
+    def raise_for_status(self, response):
+        """
+        Raise error if aiohttp response status is >= 400 and not silenced.
+        """
+        if response.status not in self.silence_errors_for_response_status_codes:
+            response.raise_for_status()
+
+        if response.status >= 400:
```

Comment:
Should this be an `else` clause because if the plugin doesn't use `silence_errors...` then I believe the aiohttp exception type should prevail?

Reply:
Also this method would benefit from a `Raises:` section describing the `FileNotFoundError` and the `aiohttp.Client...` exception being raised and which cases.

Reply:
I don't know which and how many exceptions aiohttp can raise here

Reply:
There are two. One is the `FileNotFoundError` your implementation raises. The other is the one documented here https://docs.aiohttp.org/en/stable/client_reference.html#aiohttp.ClientResponse.raise_for_status